process-create-data-repo-request

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env python3
+import re
+from datetime import datetime
+from optparse import OptionParser
+from os.path import dirname, abspath
+from os.path import expanduser
+from socket import setdefaulttimeout
+from typing import Optional, Any
+
+from categories_map import CMSSW_CATEGORIES
+
+try:
+    from yaml import CLoader as Loader, CDumper as Dumper
+except ImportError:
+    from yaml import Loader, Dumper
+from github import Github
+
+from categories import CMSSW_L2
+from cms_static import (
+    GH_CMSSW_ORGANIZATION,
+    GH_CMSSW_REPO,
+    CREATE_REPO,
+)
+from github_utils import api_rate_limits
+from os import environ
+
+setdefaulttimeout(120)
+
+JENKINS_PREFIX = "jenkins"
+try:
+    JENKINS_PREFIX = environ["JENKINS_URL"].strip("/").split("/")[-1]
+except (KeyError, IndexError):
+    JENKINS_PREFIX = "jenkins"
+
+try:
+    CMS_BOT_DIR = dirname(abspath(__file__))
+except NameError as e:
+    from sys import argv
+
+    CMS_BOT_DIR = dirname(abspath(argv[0]))
+#
+# Processes a github issue to check if it is requesting the creation of a new data repo
+# If the issue is not requesting new repo, it ignores it.
+
+# -------------------------------------------------------------------------------
+# Global Variables
+# --------------------------------------------------------------------------------
+
+APPROVAL_COMMENT = "^[+]1$"
```

Comment:
@iarspider , I think let the normal bot take care of the approval and other stuff e.g. assigning the label etc. This script should only
- assign categories (based on the data repo name) e.g. it should  just add a comment `assign xyz` and then normal bot will take care of the labels. 
- Once L2 of those categories signed (`+1`) then normal bot will add fully signed label
- The script should create the repo once issue is fully signed

Reply:
So basically it should do something like
- Thumbs up the issue description comment to indicate that bot has recognize the issue
- add comment `assign category1[,category2[,...]]` if issue has only `pending-signatures` label.
- close issue if data repo already exists (add a comment too)
- create the repository when issue has `fully-signed` label and close the issue once repo is successfully created (add a comment too)
- do nothing if issue is already closed

the jenkins job to run this should be trigger after `cms-bot` job. May be better to update the cms-bot to trigger it if the issue comment matches the regexp

Reply:
> add comment `assign category1[,category2[,...]]`
The requested repository can match multiple categories (e.g. reco and geometry), or it is always one?

>  if issue has only `pending-signatures` label.
you mean `pending-assignment`, right?

Reply:
yes a package can match multiple categories. and yes I meant `pending-assignment` :-)